agent/agent_test.go

```diff
@@ -92,9 +92,10 @@ func TestNewAgent(t *testing.T) {
 	messageEncoder := message.NewMessagesEncoder(false)
 
 	mockMetricsReporters := []metrics.Reporter{mockMetricsReporter}
+	sessionPool := session.NewSessionPool()
 
 	mockMetricsReporter.EXPECT().ReportGauge(metrics.ConnectedClients, gomock.Any(), gomock.Any())
-	ag := NewAgent(mockConn, mockDecoder, mockEncoder, mockSerializer, hbTime, 10, dieChan, messageEncoder, mockMetricsReporters)
+	ag := newAgent(mockConn, mockDecoder, mockEncoder, mockSerializer, hbTime, 10, dieChan, messageEncoder, mockMetricsReporters, sessionPool).(*agentImpl)
```

Comment:
maybe `newAgentImpl` instead of `newAgent` would be better as a function name.

Reply:
Actually, newAgent could return the concrete type instead of the interface, that way you wouldn't need to cast the type here.